clang/lib/CIR/CodeGen/CIRGenBuiltin.cpp

```diff
@@ -1649,8 +1649,11 @@ RValue CIRGenFunction::buildBuiltinExpr(const GlobalDecl GD, unsigned BuiltinID,
 
   case Builtin::BI__builtin_signbit:
   case Builtin::BI__builtin_signbitf:
-  case Builtin::BI__builtin_signbitl:
-    llvm_unreachable("BI__builtin_signbit like NYI");
+  case Builtin::BI__builtin_signbitl: {
+    mlir::Value val = buildScalarExpr(E->getArg(0));
```

Comment:
Please follow the OG codegen skeleton: create a `buildSignBit` function and put this logic there. Because we don't have PPC support yet, we need to mark it as a missing feature, so you need to add a `isPPC_FP128Ty` free function to `MissingFeatures.h` and add an assert in `buildSignBit`: `assert(!::cir::MissingFeatures::isPPC_FP128Ty());`

Reply:
I only added `MissingFeatures::isPPC_FP128Ty()` and chose not to include `buildSignBit`. 

Since we support signbit through an additional operation called `SignBitOp`, the actual processing is handled in `CIRSignBitOpLowering`. While adding the `buildSignBit` function would be quite straightforward, I felt it was best to skip it for now. Additionally, since the `builder` is a private member of `CIRGenFunction`, incorporating this function would require changing the member's access permissions to create `SignBitOp`.

Reply:
We usually try to follow the skeleton a bit closer because it makes it easier to diff against OG, if you can add it up it'd be great.

> since the builder is a private member of CIRGenFunction

You can use `CGF.getBuilder()`.